app/Repositories/Dashboard/DashboardRepository.php

```diff
@@ -0,0 +1,18 @@
+<?php
+
+namespace App\Repositories\Dashboard;
+
+use App\Models\Idea;
+
+class DashboardRepository implements IDashboardRepository 
+{
+    public function show()
+    {
+        $ideas = Idea::orderBy('created_at', 'DESC')->paginate(5);
+        if (request()->has('search'))
+        {
+            $ideas = Idea::where('content','LIKE','%'.request()->get('search', '') .'%')->orderBy('created_at', 'DESC')->paginate(5);
```

Comment:
```suggestion
            $ideas = Idea::where('content' , 'LIKE' , '%' . request()->get('search',  '')  . '%')->orderBy('created_at',  'DESC')->paginate(5);
```

Hạn chế fix cứng số trong code, e có thể khai báo biến const ở phía trên và gọi ở những chỗ dùng